oauth2_client/tests/test_flow_jwt.py

```diff
@@ -0,0 +1,177 @@
+        "scope": "web openid api id",
+        "instance_url": "https://yourInstance.salesforce.com",
+        "id": "https://yourInstance.salesforce.com/id/00Dxx0000001gPLEAY/005xx000001SwiUAAS",
+        "token_type": "Bearer"
+        }""" % self.MOCK_ACCESS_TOKEN
+
+    @requests_mock.Mocker()
+    def test_jwt_flow(self, m):
+        """
+        Test JWT Token Bearer flow fetches token
+        """
+        from oauth2_client.fetcher import fetch_token
+        from oauth2_client.tests.factories import ApplicationFactory
+
+        app_data = {
+            'authorization_grant_type': 'jwt-bearer',
+            'client_secret': self.TEST_KEY,
+            'extra_settings': {'subject': 'xyz@abx.com.lightning'},
+            'token_uri': 'https://test.salesforce.com/services/oauth2/token',
+        }
+        app = ApplicationFactory(**app_data)
+        # mock provider's response
+        m.post(
+            "https://test.salesforce.com/services/oauth2/token",
+            text=self.get_mocked_response_jwt_flow()
+        )
+
+        actual_token = fetch_token(app)
+        self.assertEqual(actual_token.token, self.MOCK_ACCESS_TOKEN)
+        self.assertEqual(set(actual_token.scope.split()), {"web", "openid", "api", "id"})
+        self.assertEqual(actual_token.token_type, "Bearer")
+
+    def test_subject_not_specified(self):
+        """
+        Raise an exception when no subject specified. Subject is always required
+        on the Application instance (in extra_settings['subject'])
+        """
+        from oauth2_client.tests.factories import ApplicationFactory
+        app_data = {
+            'authorization_grant_type': 'jwt-bearer',
+        }
+        app = ApplicationFactory(**app_data)
+        with self.assertRaises(ValidationError):
+            app.clean()  # django's interface to trigger validation
+
+    def get_mocked_response_jwt_flow_no_scope(self):
+        """
+        Mocked response with no scope, similar to the one from Salesforce
+        """
+        return """{
+        "access_token": "%s",
+        "instance_url": "https://yourInstance.salesforce.com",
+        "id": "https://yourInstance.salesforce.com/id/00Dxx0000001gPLEAY/005xx000001SwiUAAS",
+        "token_type": "Bearer"
+        }""" % self.MOCK_ACCESS_TOKEN
+
+    @requests_mock.Mocker()
+    def test_jwt_flow_no_scope(self, m):
+        """
+        Test JWT Token Bearer flow. No scope returned from provider.
+        """
+        from oauth2_client.fetcher import fetch_token
+        from oauth2_client.tests.factories import ApplicationFactory
+
+        app_data = {
+            'authorization_grant_type': 'jwt-bearer',
+            'client_secret': self.TEST_KEY,
+            'extra_settings': {'subject': 'xyz@abc.com.lightning'},
+            'token_uri': 'https://test.salesforce.com/services/oauth2/token',
+        }
+        app = ApplicationFactory(**app_data)
+        # mock provider's response
+        m.post(
```

Comment:
Please do not use 1 and 2 chars variables. Should be in pylint settings.

Reply:
Installing pylint plugin for Pycharm so we don't have to waste not time on those anymore :)